examples/wasm/main.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2019 Intel Corporation.  All rights reserved.
+ * Copyright (C) 2020 TU Bergakademie Freiberg Karl Fessel
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+//
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdint.h>
+
+#include <string.h>
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-prototypes"
+#include "wasm_export.h"
+#pragma GCC diagnostic pop
+
+#include "blob/hello.wasm.h"
+
+#define DEFAULT_THREAD_STACKSIZE (6 * 1024)
+#define DEFAULT_THREAD_PRIORITY 50
+
+#define ERROR_BUFSIZ    (128)
+#define WAMR_POOL_SIZE  (10 * 1024)
+#define WAMR_HEAP_SIZE  (2 * 1024)
+#define WAMR_STACK_SIZE (1 * 1024)
+
+static char wasm_heap[WAMR_POOL_SIZE] = { 0 };
+
+int main(void)
+{
+    char error_buf[ERROR_BUFSIZ];
+    RuntimeInitArgs init_args = { 0 };
+
+    /*
+       init_args.mem_alloc_type = Alloc_With_Pool;
+       init_args.mem_alloc_option.pool.heap_buf = wasm_heap;
+       init_args.mem_alloc_option.pool.heap_size = WAMR_POOL_SIZE;
+     */
+
+    init_args.mem_alloc_type = Alloc_With_Allocator;
+    init_args.mem_alloc_option.allocator.malloc_func = malloc;
+    init_args.mem_alloc_option.allocator.realloc_func = realloc;
+    init_args.mem_alloc_option.allocator.free_func = free;
+
+    /* initialize runtime environment */
+    if (!wasm_runtime_full_init(&init_args)) {
+        puts("wasm_runtime_full_init() failed");
+        return 1;
+    }
+
+    /* the bytecode might change, so allocate a copy */
+    uint8_t *wasm_buf = malloc(hello_wasm_len);
+
+    if (!wasm_buf) {
+        puts("out of memory");
+        goto fail1;
```

Comment:
Achievement unlocked: Go to Hell!

(For reference, see this German article ["Go to Hell" und 31 andere Erfolge für Programmierer](https://www.golem.de/1201/89181.html).)

But being honest again: Often it is easy to restructure the code to not need `goto`s by refactoring the code into smaller functions.